inference-engine/src/mkldnn_plugin/nodes/mkldnn_roi_pooling_node.cpp

```diff
@@ -677,13 +648,74 @@ void MKLDNNROIPoolingNode::execute() {
                 }
             }
 
-            if (roi_pooling_kernel) {
-                (*roi_pooling_kernel)(&arg);
+            if (execPtr) {
+                execPtr->exec(arg);
             }
         }
     });
 }
 
+void MKLDNNROIPoolingNode::executeDynamicImpl(mkldnn::stream strm) {
+    return execute(strm);
+}
+
+void MKLDNNROIPoolingNode::prepareParams() {
+    const auto& srcMemPtr0 = getParentEdgeAt(0)->getMemoryPtr();
+    const auto& srcMemPtr1 = getParentEdgeAt(0)->getMemoryPtr();
+    const auto& dstMemPtr = getChildEdgeAt(0)->getMemoryPtr();
+    if (!srcMemPtr0 || !srcMemPtr0->GetPrimitivePtr())
+        IE_THROW() << "Input memory didn't allocate.";
+    if (!srcMemPtr1 || !srcMemPtr1->GetPrimitivePtr())
+        IE_THROW() << "Input memory didn't allocate.";
+    if (!dstMemPtr || !dstMemPtr->GetPrimitivePtr())
+        IE_THROW() << "Destination memory didn't allocate.";
+    if (getSelectedPrimitiveDescriptor() == nullptr)
+        IE_THROW() << "Preferable primitive descriptor is not set.";
+
+    const auto& inDims = getParentEdgeAt(0)->getMemory().getStaticDims();
+    const auto& outDims = getChildEdgesAtPort(0)[0]->getMemory().getStaticDims();
+
+    refParams.mb = outDims[0];
+    refParams.c = rnd_up(inDims[1], refParams.c_block);
+    refParams.nb_c = refParams.c / refParams.c_block;
+    refParams.ih = inDims[2];
+    refParams.iw = inDims[3];
+    refParams.oh = outDims[2];
+    refParams.ow = outDims[3];
+
+    if (mayiuse(cpu::x64::sse41)) {
+        execPtr = std::make_shared<ROIPoolingExecutor>(refParams);
+    }
+}
+
+MKLDNNROIPoolingNode::ROIPoolingExecutor::ROIPoolingExecutor(const jit_roi_pooling_params& jpp) {
```

Comment:
Looking on current implementation I don't see any value in introducing executor abstraction. The same behavior could be achieved with pure roi_pooling_kernel usage. In good scenario we should have moved execute() body inside the executor.

Reply:
execute() body was moved to the executor. Also, Executor was separated into two parts: reference and JIT executor